### PR TITLE
Format using cfformat

### PR DESCRIPTION
**Pull request by elpete**
Also adds a check in .travis.yml to ensure files are formatted before merging

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -60,6 +60,8 @@ before_script:
   - box server start name=$ENGINE debug=true
 
 script:
+  # Check formatting
+  - box run-script format:check
```
Comment (lmajano) [inline .travis.yml]: This actually breaks the current file sets as not all of them adhere to it.  Can you remove this for now.

Comment (elpete) [reply .travis.yml]: That would miss the entire point which is to not let bad formatting get back in the code base.  If we don't have this check we will constantly be having formatting commits like this.

I've adjusted some of the formatting scripts, but the point is they _should_ all pass the `format:check`.

**Comment by elpete** [comment]
@lmajano As you can see, the check passed here.

File: system/Bootstrap.cfc [hunk 2]
```diff
@@ -1,719 +1,795 @@
...
+    property name="COLDBOX_APP_MAPPING";
+    // Lock Timeout for startup operations
+    property name="lockTimeout";
+    // The application hash used for locks
+    property name="appHash";
+    // By default if an app is reiniting and a request hits it, we will fail fast with a message
+    property name="COLDBOX_FAIL_FAST";
+
+    // param the properties with defaults
+    param name="COLDBOX_CONFIG_FILE" default="";
```
Comment (lmajano) [inline system/Bootstrap.cfc]: @elpete I have some concerns with the auto-formmatting that maybe we can involve John.  I am marking them here. Basically, the auto formatting is loosing my code alignments

File: system/Bootstrap.cfc [hunk 3]
```diff
@@ -1,719 +1,795 @@
...
+        // Setup the Framework And Application
+        application[ appKey ]
+            .getLoaderService()
+            .loadApplication( COLDBOX_CONFIG_FILE, COLDBOX_APP_MAPPING );
+        // Application Start Handler
+        try{
+            if( len( application[ appKey ].getSetting( "ApplicationStartHandler" ) ) ){
+                application[ appKey ].runEvent( event = application[ appKey ].getSetting( "ApplicationStartHandler" ) );
+            }
+        }catch( any e ){
```
Comment (lmajano) [inline system/Bootstrap.cfc]: I thought we had a rule to space out these things too?

File: system/Bootstrap.cfc [hunk 4]
```diff
@@ -1,719 +1,795 @@
...
+                                .getInterceptorService()
+                                .processState( "preReinit" );
+                            // Shutdown the application services
+                            application[ appKey ].getLoaderService().processShutdown();
+                        }
+                        // Reload ColdBox
+                        loadColdBox();
+                        // Remove any context stragglers
+                        structDelete( request, "cb_requestContext" );
+                    }catch( any e ){
```
Comment (lmajano) [inline system/Bootstrap.cfc]: Same here

File: system/Bootstrap.cfc [hunk 5]
```diff
@@ -1,719 +1,795 @@
...
+                            // Discover from event caching first.
+                            if( !structIsEmpty( renderData ) ){
+                                defaultContentType = renderData.contentType;
+                            }else{
+                                // Else, ask the engine
+                                defaultContentType = getPageContextResponse().getContentType();
+                            }
+
+                            // prepare storage entry
+                            var cacheEntry = {
```
Comment (lmajano) [inline system/Bootstrap.cfc]: My formatting here is lost, where the entries where aligned and spaced?

File: system/Bootstrap.cfc [hunk 6]
```diff
@@ -1,719 +1,795 @@
...
+            // Closure or UDF
+            if( isClosure( variables.COLDBOX_FAIL_FAST ) || isCustomFunction( variables.COLDBOX_FAIL_FAST ) ){
+                variables.COLDBOX_FAIL_FAST();
+                return false;
+            }
+            // Core Fail Fast Option
+            else if( isBoolean( variables.COLDBOX_FAIL_FAST ) && variables.COLDBOX_FAIL_FAST ){
+                writeOutput( "Oops! Seems ColdBox is still not ready to serve requests, please try again." );
+                // You don't have to return a 500, I just did this so JMeter would report it differently than a 200
+                cfheader(statusCode="503", statustext="ColdBox Not Available Yet!");
```
Comment (lmajano) [inline system/Bootstrap.cfc]: Spacing is not used here but it was before?

File: .cfformat.json [hunk 7]
```diff
@@ -1,29 +1,53 @@
...
\ No newline at end of file
+    "strings.attributes.quote":"double",
+    "strings.quote":"double",
+    "struct.empty_padding":false,
+    "struct.multiline.element_count":4,
+    "struct.multiline.leading_comma":false,
+    "struct.multiline.leading_comma.padding":true,
+    "struct.multiline.min_length":40,
+    "struct.padding":true,
+    "struct.separator":": ",
```
Comment (lmajano) [inline .cfformat.json]: we need some spacing here

**Comment by elpete** [comment]
I'm planning to close this out and open another PR because of the conflicts.  Where are we at with the library and @jcberquist?  Do we need any other adjustments before we can do this?

**Comment by lmajano** [comment]
I am a little hesitant on running format on ALL files.

**Comment by elpete** [comment]
If you are hesitant, let's solve what is making you hesitant.  Let's not let it stop us from automating this.

**Comment by lmajano** [comment]
@elpete I have merged this locally as lots of stuff needed updating locally. I am formatting packages little by little. But the scripts have been added. I will close for now. Thanks! THis was a big motivational ticket!

**Pull request closed without merging**